packer_test/common/check/file_gadgets.go

```diff
@@ -35,31 +35,24 @@ func FileExists(filePath string, isDir bool) Checker {
 	}
 }
 
-type fileInDir struct {
-	filename string
-	dirPath  string
+type fileGlob struct {
+	filepath string
 }
 
-func (fe fileInDir) Check(_, _ string, _ error) error {
-	files, err := os.ReadDir(fe.dirPath)
+func (fe fileGlob) Check(_, _ string, _ error) error {
+	matches, err := filepath.Glob(fe.filepath)
 	if err != nil {
-		return fmt.Errorf("failed to read dir %q: %s", fe.dirPath, err)
+		return fmt.Errorf("error finding file %q: %q", fe.filepath, err)
 	}
 
-	pattern := regexp.MustCompile(fe.filename)
-
-	for _, file := range files {
-		if !file.IsDir() && pattern.MatchString(file.Name()) {
-			return nil
-		}
+	if len(matches) == 0 {
+		return fmt.Errorf("file %q not found", fe.filepath)
```

Comment:
I would amend this message to more clearly explicit that the glob didn't match anything here.
Something like `no matches found for glob expression %q`?